Library/Homebrew/utils/analytics.sh

```diff
@@ -1,30 +1,42 @@
 setup-analytics() {
-  # User UUID file. Used for Homebrew user counting. Can be deleted and
-  # recreated with no adverse effect (beyond our user counts being inflated).
+  HOMEBREW_GIT_CONFIG_FILE="$HOMEBREW_REPOSITORY/.git/config"
+
+  # User UUID file. This file has been deprecated in favor of
+  # homebrew.analyticsuuid in $HOMEBREW_REPOSITORY/.git/config. If detected,
+  # the UUID will be written to the aforementioned git config file, and the
+  # original UUID file will be removed.
   HOMEBREW_ANALYTICS_USER_UUID_FILE="$HOME/.homebrew_analytics_user_uuid"
+  if [[ -f "$HOMEBREW_ANALYTICS_USER_UUID_FILE" ]]
+  then
+    HOMEBREW_ANALYTICS_USER_UUID="$(<"$HOMEBREW_ANALYTICS_USER_UUID_FILE")"
+    if [[ -n "$HOMEBREW_ANALYTICS_USER_UUID" ]]
+    then
+      git config --file="$HOMEBREW_GIT_CONFIG_FILE" --replace-all homebrew.analyticsuuid "$HOMEBREW_ANALYTICS_USER_UUID"
+    fi
+    rm -f "$HOMEBREW_ANALYTICS_USER_UUID_FILE"
+  fi
 
   # Make disabling anlytics sticky
   if [[ -n "$HOMEBREW_NO_ANALYTICS" ]]
   then
-    git config --file="$HOMEBREW_REPOSITORY/.git/config" --replace-all homebrew.analyticsdisabled true
+    git config --file="$HOMEBREW_GIT_CONFIG_FILE" --replace-all homebrew.analyticsdisabled true
     # Internal variable for brew's use, to differentiate from user-supplied setting
     export HOMEBREW_NO_ANALYTICS_THIS_RUN="1"
   fi
 
-  if [[ "$(git config --file="$HOMEBREW_REPOSITORY/.git/config" --get homebrew.analyticsmessage)" != "true" ||
-        "$(git config --file="$HOMEBREW_REPOSITORY/.git/config" --get homebrew.analyticsdisabled)" = "true" ]]
+  if [[ "$(git config --file="$HOMEBREW_GIT_CONFIG_FILE" --get homebrew.analyticsmessage)" != "true" ||
+        "$(git config --file="$HOMEBREW_GIT_CONFIG_FILE" --get homebrew.analyticsdisabled)" = "true" ]]
```

Comment:
I'll be conservative and stick with git commands then, because I don't think `funcname --replace-all` will be very readable. One optimization would be to set `GIT_CONFIG` at the beginning of `setup-analytics` and unset it at the end, so we don't need to have `--file` everywhere.